addons/survey/models/survey.py

```diff
@@ -0,0 +1,1159 @@
+        for label in self:
+            # 'bool()' is required in order to make '!=' act as XOR with objects
+            if not bool(label.question_id) != bool(label.question_id_2):
+                raise UserError(_("A label must be attached to one and only one question"))
+
+class SurveyUserInput(models.Model):
+    ''' Metadata for a set of one user's answers to a particular survey '''
+    _name = "survey.user_input"
+    _rec_name = 'date_create'
+    _description = 'Survey User Input'
+
+    survey_id = fields.Many2one('survey.survey', string='Survey', required=True,
+                                 readonly=True, ondelete='restrict')
+    date_create = fields.Datetime(string='Creation Date', required=True,
+                                   readonly=True, copy=False, default=fields.Datetime.now)
+    deadline = fields.Datetime(string="Deadline",
+                            help="Date by which the person can open the survey and submit answers",
+                            oldname="date_deadline")
+    input_type = fields.Selection([('manually', 'Manually'), ('link', 'Link')],
+                             string='Answer Type', required=True, readonly=True,
+                             oldname="response_type", default='manually')
+    state = fields.Selection([('new', 'Not started yet'),
+                               ('skip', 'Partially completed'),
+                               ('done', 'Completed')],
+                              string='Status',
+                              readonly=True, default='new')
+    test_entry = fields.Boolean(readonly=True)
+    token = fields.Char(string="Identification token", readonly=True, required=True, copy=False, default=lambda self: uuid.uuid4().__str__())
+
+    # Optional Identification data
+    partner_id = fields.Many2one('res.partner', string='Partner', readonly=True)
+    email = fields.Char(string="E-mail", readonly=True)
+
+    # Displaying data
+    last_displayed_page_id = fields.Many2one('survey.page',
+                                          string='Last displayed page')
+    # The answers !
+    user_input_line_ids = fields.One2many('survey.user_input_line',
+                                           'user_input_id', string='Answers', copy=True)
+
+    # URLs used to display the answers
+    result_url = fields.Char(related='survey_id.result_url',
+                                 string="Public link to the survey results")
+    print_url = fields.Char(related='survey_id.print_url',
+                                string="Public link to the empty survey")
+
+    quizz_score = fields.Float(compute="_compute_quizz_score", string="Score for the quiz")
+
+    _sql_constraints = [
+        ('unique_token', 'UNIQUE (token)', 'A token must be unique!'),
+        ('deadline_in_the_past', 'CHECK (deadline >= date_create)', 'The deadline cannot be in the past')
+    ]
+
+    @api.multi
+    def _compute_quizz_score(self):
+        for user_input in self:
```

Comment:
we have group_operator in field. I think, default group_operator is "sum". 

so read_group of "input line" will be return sum(quizz_mark) group by input (per input).